crates/pet-conda/src/lib.rs

```diff
@@ -147,73 +148,81 @@ impl Locator for Conda {
         None
     }
 
-    fn find(&self) -> Option<LocatorResult> {
-        // 1. Get a list of all know conda environments
-        let known_conda_envs =
-            get_conda_environments(&get_conda_environment_paths(&self.env_vars), &None);
-        let mut new_managers = vec![];
-        {
-            let mut managers = self.managers.lock().unwrap();
-            // 2. Go through all conda dirs and build the conda managers.
-            for env in &known_conda_envs {
-                if let Some(conda_dir) = &env.conda_dir {
-                    if managers.contains_key(conda_dir) {
-                        continue;
-                    }
-                    if let Some(manager) = CondaManager::from(conda_dir) {
-                        new_managers.push(manager.to_manager());
-                        managers.insert(conda_dir.clone(), manager);
+    fn find(&self, reporter: &dyn Reporter) {
+        let env_vars = self.env_vars.clone();
+        thread::scope(|s| {
+            // 1. Get a list of all know conda environments file paths
+            let possible_conda_envs = get_conda_environment_paths(&env_vars);
+            for path in possible_conda_envs {
+                s.spawn(move || {
+                    // 2. Get the details of the conda environment
+                    // This we do not get any details, then its not a conda environment
+                    let env = get_conda_environment_info(&path, &None)?;
+
+                    // 3. If we have a conda environment without a conda_dir
+                    // Then we will not be able to get the manager.
+                    // Either way report this environment
+                    if env.conda_dir.is_none(){
+                        // We will still return the conda env even though we do not have the manager.
+                        // This might seem incorrect, however the tool is about discovering environments.
+                        // The client can activate this env either using another conda manager or using the activation scripts
+                        error!("Unable to find Conda Manager for the Conda env: {:?}", env);
+                        let prefix = env.prefix.clone();
+                        let env = env.to_python_environment(None, None);
+                        let mut environments = self.environments.lock().unwrap();
+                        environments.insert(prefix, env.clone());
+                        reporter.report_environment(&env);
```

Comment:
as soon as we discover anything we report it.
Previously we-d collect and return to calling code.